docs/cluster_api.md

```diff
@@ -93,3 +80,37 @@ In more detail:
   - LastTransitionTime [string]
   - Message [string]
   - Reason [string]
+
+### MachineSetConfig
+  - `NodeType [NodeType:string]` *Enumerated node type for the machine set (compute or master)*
+  - `Infra [boolean]` *True if this machine set should contain infrastructure pods (registry, router)*
+  - `Size [int]` *How many machines should be created for this machine set*
+  - `Version [corev1.LocalObjectReference]` *Reference to ClusterVersion resource*
+  - `Hardware`
+    - `AWS` *-optional- contains AWS specific configuration*
+      - `InstanceType [string]` *EC2 instance type (ie. m4.xlarge)*
+  - `NodeLabels [map[string]string]` *Labels to apply to nodes created in this MachineSet*
+
+### ClusterVersion
+Standalone resource that encapsulates details of a named version. It
+defines general OpenShift details about a particular version such as 
+the image template and RPM version as well as cloud-specific information
+such as the AWS AMI name.
+
+A set of predefined ClusterVersion resources can be initially created in a shared 
+namespace (like `openshift`) to define the versions supported out of the box.
+Users are able to create their own in their own namespace and associate their
+clusters with that.
+
+- `ObjectMetadata (inline) [metav1.ObjectMetadata]`
+- `TypeMetadata (inline) [metav1.TypeMetadata]`
+
+#### Spec
+- `ImageURL [string]` *-optional- template for images to be used on the cluster (oreg_url in ansible)*
+- `ImageTag [string]` *-optional- the image tag for OpenShift images (either ImageURL or ImageTag can be specified, but not both, openshift_image_tag in ansible)*
+- `Release [string]` *-optional- simple release name (openshift_release in ansible)*
```

Comment:
I'm unsure if we're going to need this one given how things are playing out. This was pretty focused on the end customer using ansible to just install the latest v3.x, in our world with pre-defined versions including specific AMIs, we might just want to stay specific and skip this one.